### PR TITLE
[[ CommandArguments ]] Implementation of 2 new functions

### PR DESCRIPTION
**Pull request by livecodesebastien**
- commandArguments: returns a 1-based array of the commandline arguments, or the argument at the index (if an index is specified).
- commandName: returns the command name.

That supersedes https://github.com/runrev/livecode/pull/2606 (as these features won't make it in LiveCode 6.7).

### CODE REVIEW

File: engine/src/globals.cpp [hunk 1]
```diff
@@ -832,6 +836,18 @@ void X_clear_globals(void)
 
     MCSocketsInitialize();
     
+    if (MCcommandarguments != NULL)
```
Comment (runrevmark) [inline engine/src/globals.cpp]: This is not correct. The X_clear_globals function is to reset global variables to their initial values - when this is called they should be assumed to be unset. (It is to work around the non-reinit of zero-init data in the android engine on app restart).

All you need to do is: MCcommandarguments / MCcommandname = NULL.

Comment (livecodesebastien) [inline engine/src/globals.cpp]: I wasn't sure about this - now fixed

File: engine/src/globals.cpp [hunk 2]
```diff
@@ -928,8 +935,27 @@ bool X_open(int argc, MCStringRef argv[], MCStringRef envp[])
...
+    MCcommandarguments = NULL;
+    if (MCModeHasCommandLineArguments())
+    {
+        MCcommandname = MCValueRetain(argv[0]);
+
+        bool t_success;
+        t_success = MCArrayCreateMutable(MCcommandarguments);
+
+        // We build a 1-based numeric array.
+        for (int i = 1; t_success && i < argc; i++)
```
Comment (peter-b) [inline engine/src/globals.cpp]: The loop variable should be an `index_t`, to match the type of the argument to `MCArrayStoreValueAtIndex()`.

**Comment by peter-b** [comment]
@livecode-vulcan review ok 559a334

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok 559a334eef10a67780fecf297a8cb0a97c4a4621

**Comment by livecode-vulcan** [comment]
😎 test success 559a334eef10a67780fecf297a8cb0a97c4a4621
- try-linux-x86: [success](https://vulcan.livecode.com/builders/try-linux-x86/builds/612)
- try-win-x86: [success](https://vulcan.livecode.com/builders/try-win-x86/builds/672)
- try-ios: [success](https://vulcan.livecode.com/builders/try-ios/builds/561)
- try-android-armv6: [success](https://vulcan.livecode.com/builders/try-android-armv6/builds/625)
- try-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-linux-x86_64/builds/557)
- try-mac: [success](https://vulcan.livecode.com/builders/try-mac/builds/550)

**Pull request merged**